lighthouse-core/gather/gather-runner.js

```diff
@@ -167,6 +167,34 @@ class GatherRunner {
     }
   }
 
+  /**
+   * Returns an error if we ended up on the `chrome-error` page and all other requests failed.
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @return {LH.LighthouseError|undefined}
+   */
+  static getInterstitialError(networkRecords) {
+    const interstitialRequest = networkRecords
+      .find(record => record.documentURL.startsWith('chrome-error://'));
+    // If the page didn't end up on a chrome interstitial, there's no error here.
+    if (!interstitialRequest) return undefined;
+
+    const pageNetworkRecords = networkRecords
+      .filter(record => !URL.NON_NETWORK_PROTOCOLS.includes(record.protocol) &&
+        !record.documentURL.startsWith('chrome-error://'));
+    // If at least one request didn't fail, there's no error here.
+    if (!pageNetworkRecords.some(record => record.failed)) return undefined;
```

Comment:
what happens if you redirect to a bad page?

Reply:
then we get the initial request succeeding with the later network records failing